test/simplify_algebra_test.cpp

```diff
@@ -4133,6 +4134,132 @@ TEST_CASE(dot_slice_a)
+        auto slice_c = m1.add_instruction(
+            migraphx::make_op("slice", {{"axes", {2}}, {"starts", {1536}}, {"ends", {2304}}}), a);    
+
+        auto one = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 3.0f));
+        auto two = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 2.0f));
+        auto three = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+
+        auto add1 = m1.add_instruction(migraphx::make_op("add"), slice_a, one);
+        
+        auto add2 = m1.add_instruction(migraphx::make_op("add"), slice_b, two);
+        auto add3 = m1.add_instruction(migraphx::make_op("add"), slice_c, three);
+
+        m1.add_return({add1});
+        
+    };
+    
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto a = m2.add_parameter("a", as); 
+
+        auto one = m2.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 3.0f));
+        auto two = m2.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 2.0f));
+        auto three = m2.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+
+        auto concat = m2.add_instruction(migraphx::make_op("concat", {{"axis", 2}}), one, two, three);
+
+        auto add = m2.add_instruction(migraphx::make_op("add"), a, concat);
+
+        auto slice_a = m2.add_instruction(
+            migraphx::make_op("slice", {{"axes", {2}}, {"starts", {0}}, {"ends", {768}}}), add);
+        
+        m2.add_return({slice_a});
+    };
+    
+    EXPECT(m1.sort() == m2.sort());
+}
+
+TEST_CASE(add_dot_add_mul)
+{
+    migraphx::shape as{migraphx::shape::float_type, {1, 77, 768}};
+
+    migraphx::module m1;
+    {
+        auto a = m1.add_parameter("a", as); 
+
+        auto one = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 3.0f));
+        auto two = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 2.0f));
+        auto three = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 77, 768}}, 4.0f));
+        auto four = m1.add_literal(migraphx::generate_literal(
+            {migraphx::shape::float_type, {1, 768, 768}}, 4.0f));
+
+
+        auto add1 = m1.add_instruction(migraphx::make_op("add"), a, one);
+        auto dot1 = m1.add_instruction(migraphx::make_op("dot"), add1, four);
+        
+        auto add2 = m1.add_instruction(migraphx::make_op("add"), dot1, two);
+        auto mul = m1.add_instruction(migraphx::make_op("mul"), add2, three);
+
+        m1.add_return({mul});    
+    };
+    run_pass(m1);
+    migraphx::module m2 = m1;
+    EXPECT(m1.sort() == m2.sort());
```

Comment:
This is a problem. The add_mul should be rewritten.

Reply:
The slice_add_mul testcase is checking the case when there is a slice->add->mul, but the other slices are not followed by an add.